src/test/scala/com/mongodb/spark/TestHelper.scala

```diff
@@ -16,28 +16,51 @@
 
 package com.mongodb.spark
 
+import com.mongodb.client.model.{UpdateOptions, Updates}
+import com.mongodb.client.{MongoCollection, MongoDatabase}
+import com.mongodb.{MongoClient, MongoClientURI, ReadPreference}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.{SparkConf, SparkContext}
+import org.bson.{BsonDocument, BsonString, Document}
+
 import scala.collection.JavaConverters._
 import scala.collection.immutable.IndexedSeq
 import scala.io.Source
 import scala.util._
-import org.apache.spark.SparkConf
-import org.bson.{BsonDocument, BsonString, Document}
-import com.mongodb.client.model.{UpdateOptions, Updates}
-import com.mongodb.client.{MongoCollection, MongoDatabase}
-import com.mongodb.{Block, MongoClient, MongoClientURI, ReadPreference}
 
-object MongoDBDefaults {
-  def apply(): MongoDBDefaults = new MongoDBDefaults()
+object TestHelper {
+  def apply(): TestHelper = new TestHelper()
+
+  private var _sparkContext: Option[SparkContext] = None
+  private var customConf: Boolean = false
+
+  def getOrCreateSparkContext(sparkConf: SparkConf, requiredCustomConf: Boolean): SparkContext = synchronized {
```

Comment:
SparkContext is now created here only. If a custom SparkConf is required the old version is reset / discarded and a new one created.